src/appFunctions.js

```diff
@@ -93,6 +93,18 @@ export function getParentCardByIdChildren(id, cards) {
     return card;
 }
 
+export function getParentCardArrayByIdChildren(id, cards) {
+    let array = [];
+    let parent = findCardById(id, cards);
+    
+    while(parent) {
+        array = [parent, ...array];
+        parent = getParentCardByIdChildren(parent._id, cards);
+    }
+
+    return array;
+}
+
```

Comment:
why do you need it? it is possible to have only one parent and restrict user to add child/parent to prevent cycling dependecies
for example: c1 is parent of c2, c2 is parent of c3, but c3 can't be parent of c1 or c2 (you need to check it and prevent)

Reply:
This function returns an array of parent cards in the order of their dependency. Whatever the closed looping, I prevent the possibility of adding a child. This is checked in the function

appFunctions.js
export function findAvalibleChildsToAdd(id, cards) {
    const generalArrayCards = Object.values(cards).reduce((prev, item) => [...prev, ...item]);
    
    return generalArrayCards.filter((card) => {
        if ((id === card._id) || card.children.includes(id)) {
            return false;
        }
        
        return !generalArrayCards.find((item) => (
            item.children.includes(card._id)
        ));      
    });
}

Reply:
you are right, it's ok